mathics/builtin/atomic/strings.py

```diff
@@ -851,6 +869,16 @@ def apply_default(self, value, evaluation, options):
     def apply_form(self, expr, form, evaluation, options):
         "ToString[expr_, form_, OptionsPattern[ToString]]"
         encoding = options["System`CharacterEncoding"]
+
+        if isinstance(expr, String):
+            # ToString[] of a string parses the string into an M-expression and then
+            # runs ToString[] on the resulting M-expression.
+            session = MathicsSession()
+            try:
+                expr = parse(session.definitions, MathicsSingleLineFeeder(expr.value))
```

Comment:
really do you want to create a session to convert a string?

Reply:
I suppose it is better to just Definitions. I am not sure though if we want `add_builtin` set `True` or `False`. You tell me.

The thought here was to use a  clean slate of the definitions, and not modification made to the built-in functions. 

But you tell me, does `ToString` behavior of an expression change if there has been an altered the behavior of functions?

Reply:
> MakeBoxes does not use that information to build a Boxed expression. 

I wasn't intending to imply that. Instead there is a function that accepts an encoding parameter. In the course of doing its thing it triggers a `MakeBox`  

Right now `MakeBoxes` performs some formatter before it returns  We need to get the information through to the formatter that `MakeBoxes` is invoking .  In a functional approach which is what would be preferred here, this would passed via the function call arguments somehow. 

> It happens afterwards (when Boxes are converted into text strings)

Currently, in Mathics Boxing and Formatting can be interleaved. So right now there is no absolute "afterwards".  

In Wolfram Alpha are there ever situations where Boxing and Formatting are interleaved? 

Removing the interleaving would require a major refactoring of our code. If that needs to be done, I think we need to reduce our goals for the time being and set a more achievable bar to reach.